server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1809,16 +2126,32 @@ public static ClusterState updateWithSnapshots(ClusterState state,
      * @param snapshots    snapshots in progress
      * @return shard ids that currently have an actively executing shard snapshot on a data node
      */
-    private static Set<ShardId> busyShardsForRepo(String repoName, @Nullable SnapshotsInProgress snapshots) {
+    private static Set<ShardId> busyShardsForRepo(String repoName, @Nullable SnapshotsInProgress snapshots, Metadata metadata) {
         final List<SnapshotsInProgress.Entry> runningSnapshots = snapshots == null ? List.of() : snapshots.entries();
         final Set<ShardId> inProgressShards = new HashSet<>();
         for (SnapshotsInProgress.Entry runningSnapshot : runningSnapshots) {
             if (runningSnapshot.repository().equals(repoName) == false) {
                 continue;
             }
-            for (ObjectObjectCursor<ShardId, ShardSnapshotStatus> shard : runningSnapshot.shards()) {
-                if (shard.value.isActive()) {
-                    inProgressShards.add(shard.key);
+            if (runningSnapshot.source() == null) {
```

Comment:
It's use everywhere, maybe adding a method `isClone()` returning `source != null` would be more explicit?

Reply:
++  added that, much nicer now :)